src/tensors/cpu/ruy_adapter.h

```diff
@@ -0,0 +1,468 @@
+      // Set all registers in lane from same scalar value.
+      float32x4_t multiplier = vdupq_n_f32(unquant_multiplier);
+      const int32x4_t *Input = reinterpret_cast<const int32x4_t *>(input);
+      const int32x4_t *InputEnd = reinterpret_cast<const int32x4_t *>(input + rows_A * cols_B);
+      float32x4_t *Output = reinterpret_cast<float32x4_t *>(output);
+
+      while(Input != InputEnd) {
+        // Bias cycles every column for addition.
+        const float32x4_t *Bias = reinterpret_cast<const float32x4_t *>(input_bias_prepared);
+
+        // InputEnd needs to be determined to end the while loop below.
+        const int32x4_t *RowEnd = reinterpret_cast<const int32x4_t *>(
+            reinterpret_cast<const int32_t *>(Input) + cols_B);
+
+        while(Input != RowEnd) {
+          // Operation happening for 4-elements together:
+          // output = [int32_t]input * [float]quant_mult + [float]bias;
+          float32x4_t floatInput = vcvtq_f32_s32(*Input++);
+          float32x4_t unquantized = vmulq_f32(floatInput, multiplier);
+          *Output++ = vaddq_f32(unquantized, *Bias++);
+        }
+      }
+    }
+
+    float unquant_multiplier;
+    const float *input_bias_prepared;
+  };
+
+  struct UnquantizeAndWrite {
+    UnquantizeAndWrite(float unquant_multiplier) : unquant_multiplier(unquant_multiplier) {}
+
+    void operator()(const int32_t *input, Index rows_A, Index cols_B, float *output) {
+      // Set all registers in lane from same scalar value.
+      float32x4_t multiplier = vdupq_n_f32(unquant_multiplier);
+      const int32x4_t *Input = reinterpret_cast<const int32x4_t *>(input);
+      const int32x4_t *InputEnd = reinterpret_cast<const int32x4_t *>(input + rows_A * cols_B);
+      float32x4_t *Output = reinterpret_cast<float32x4_t *>(output);
+
+      while(Input != InputEnd) {
+        // Bias cycles every column for addition.
+
+        // InputEnd needs to be determined to end the while loop below.
+        const int32x4_t *RowEnd = reinterpret_cast<const int32x4_t *>(
+            reinterpret_cast<const int32_t *>(Input) + cols_B);
+
+        while(Input != RowEnd) {
+          // Operation happening for 4-elements together:
+          // output = [int32_t]input * [float]quant_mult + [float]bias;
+          float32x4_t floatInput = vcvtq_f32_s32(*Input++);
+          float32x4_t unquantized = vmulq_f32(floatInput, multiplier);
+          *Output++ = unquantized;
+        }
+      }
+    }
+
+    float unquant_multiplier;
+  };
+};
+
+#endif
+
+/*
+ * The following nomenclature comes from intgemm. The current state of code is to keep the
+ * intgemm_interface.h diff minimal. There are possibly better abstractions.
+ */
+struct IntgemmViaRuy {
+  // Convert compile time errors into run-time ABORTS. This allows bringing in only int8_t and
+  // select functions that are required to create a path which will run while not achieving
+  // parity with intgemm.
+  template <class T>
+  struct IntBase {
```

Comment:
Given every method is overridden, maybe just make this an Int16Unimplemented class and remove the `IntBase` inheritance?

Reply:
`PrepareB` is unimplemented as of now (I think I'm getting away with this because `PrepareBTransposed` is being used at runtime).

Reply:
I have removed inheritance, essentially moving Unimplemented to Int16 and copying prepareB unimplemented into int8